libopenage/gamestate/resource.h

```diff
@@ -46,6 +46,9 @@ class ResourceProducer : public Resource {
  * All the resources.
  *
  * The the ids of the resources must be inside [0, count).
+ *
+ * The static variables wood, food, gold, stone are the ids of the representing resource.
+ * Any extension of Resources must use this ids as they are an engine dependency (at the moment).
```

Comment:
@TheJJ Do you like this? Could also be done for all dependencies on the data to avoid strings ( `Resources::food` instead of `resources.get("openage.Food")` ) ( I hate strings )

Reply:
At the end there should only be one eg. `Resource::money` that will be used for the market like stuff and it it will be mapped to gold for the classic data

Reply:
Nyan impl:
```python
Resources()
  resources: orderedset(Resource)
  money: Resource
````
If `money` is missing make money the first resource in the set

Reply:
I don't like strings either but the non-string access should just be an optimization to the string-access (for simplicity and consistency and scriptability). That means that if we should store arbitrary resources by their identifier in a dynamic data structure and relay accesses to it to either this dynamic storage or direct class members. That way the string access can be done anyway (e.g. from pythonscripts) but the engine can access the member directly in the struct memory.

The engine API would just provide a generic `Resource`, and the base game would then instanciate this for `Gold`, `Food`, `Wood`, `Stone`.

Having some definition for which resource is the money is really great, we should do that :)
This means we then set `money = Gold`.

`nyan` can enforce that variables must have values, so `money : Resource` must be set by the base game data mod, otherwise nyan will say "undefined member". We can tweak that behavior of course, if needed.

Reply:
Additionally, except the money thing, the core engine probably never needs to know what the resources are and then do the direct-member-access. The HUD just displays the resource set, and the engine instanciates resource storage for each player, depending on the contents of the set.